src/nccl_ofi_interface_nvidia.c

```diff
@@ -318,3 +318,37 @@ NCCL_OFI_EXPORT_SYMBOL const ncclNet_v8_t ncclNetPlugin_v8 = {
         .getDeviceMr = NULL,
         .irecvConsumed = NULL,
 };
+
+
+/*
+ * Versions 1.11.0 and prior of the plugin set the name to
+ * "AWS Libfabric", requiring NCCL_NET be set to "AWS Libfabric",
+ * opening the door to shell escape failures.  Customers do have
+ * NCCL_NET="AWS Libfabric" in their various scripts, so still support
+ * that.  And, since we're here, also deal with the constant
+ * "Libfabric" vs. "OFI" confusion.
+ */
+__attribute__((constructor)) static void nvidia_plugin_name_fixup(void)
+{
+	char *net_env;
+
+	net_env = getenv("NCCL_NET");
+	if (net_env != NULL && 0 == strncasecmp(net_env, "AWS Libfabric", strlen("AWS Libfabric"))) {
```

Comment:
Nit: `strncasecmp` feels redundant here compared to `strcasecmp` -- the latter saves the third parameter.